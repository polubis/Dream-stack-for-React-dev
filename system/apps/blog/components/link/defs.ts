import type { ReactNode } from 'react';

interface LinkProps {
  href: string;
  children: ReactNode;
  title: string;
}

export type { LinkProps };
