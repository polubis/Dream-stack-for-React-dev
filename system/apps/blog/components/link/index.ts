export * from './defs';
export * from './link';
