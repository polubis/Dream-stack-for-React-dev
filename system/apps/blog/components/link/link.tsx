import NextLink from 'next/link';
import type { LinkProps } from './defs';

const Link = ({ href, children }: LinkProps) => {
  return <NextLink href={href}>{children}</NextLink>;
};

export { Link };
