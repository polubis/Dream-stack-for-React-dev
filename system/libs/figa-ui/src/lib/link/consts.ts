const LINK_MOTIVES = ['default', 'primary'] as const;

export { LINK_MOTIVES };
