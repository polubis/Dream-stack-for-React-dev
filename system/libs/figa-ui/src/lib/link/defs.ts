import type { ReactNode } from 'react';
import type { FontVariant } from '../font';

interface LinkProps {
  className?: string;
  variant: FontVariant;
  children: ReactNode;
}

export type { LinkProps };
