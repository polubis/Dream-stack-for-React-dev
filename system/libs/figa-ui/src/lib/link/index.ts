export * from './link';
export * from './defs';
