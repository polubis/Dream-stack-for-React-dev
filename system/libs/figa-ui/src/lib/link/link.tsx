import type { LinkProps } from './defs';

import c from 'classnames';
import { Font } from '../font';

const Link = ({ className, variant, children }: LinkProps) => {
  return (
    <Font className={c('link', className)} variant={variant} element="span">
      {children}
    </Font>
  );
};

export { Link };
