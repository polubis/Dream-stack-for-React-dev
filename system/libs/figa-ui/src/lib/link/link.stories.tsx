import type { Story, Meta } from '@storybook/react';

import { Link } from './link';
import { FONT_VARIANTS } from '../font';

export default {
  component: Link,
  title: 'Link',
} as Meta;

const Template: Story = () => {
  return (
    <>
      {FONT_VARIANTS.map((variant) => (
        <div key={variant} style={{ margin: '8px 0' }}>
          <Link variant={variant}>
            <a href="/">{variant}</a>
          </Link>
        </div>
      ))}
    </>
  );
};

export const Default = Template.bind({});
Default.args = {};
