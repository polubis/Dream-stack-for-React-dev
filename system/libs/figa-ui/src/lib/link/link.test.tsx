import { render, screen } from '@testing-library/react';
import { Link } from './link';

describe('link can be used when: ', () => {
  it('[FRAGILE] renders with default props', () => {
    const { asFragment } = render(
      <Link variant="h5">
        <a href="/">Link</a>
      </Link>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('[FRAGILE] allows to pass classes', () => {
    render(
      <Link className="class1 class2" variant="h2">
        <a href="/">Link</a>
      </Link>
    );

    const link = screen.getByText(/Link/);
    expect(link.parentElement?.className).toContain('class1 class2');
  });

  it('[FRAGILE] the parent node is a span', () => {
    render(
      <Link variant="h2">
        <a href="/">Link</a>
      </Link>
    );

    const link = screen.getByText(/Link/);
    expect(link.parentElement?.tagName).toBe('SPAN');
  });
});
