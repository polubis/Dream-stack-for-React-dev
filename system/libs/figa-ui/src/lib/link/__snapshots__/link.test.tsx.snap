// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`link can be used when:  [FRAGILE] renders with default props 1`] = `
<DocumentFragment>
  <span
    class="font font-h5 link"
  >
    <a
      href="/"
    >
      Link
    </a>
  </span>
</DocumentFragment>
`;
